cpp/src/arrow/type.h

```diff
@@ -456,7 +456,7 @@ struct ARROW_EXPORT Date32Type : public DateType {
 
   Date32Type();
 
-  int bit_width() const override { return static_cast<int>(sizeof(c_type) * 4); }
+  int bit_width() const override { return static_cast<int>(sizeof(c_type) * 8); }
```

Comment:
`CHAR_BIT` is defined in `<climits>` and I don't think `std::numeric_limits` has an analog.